scalameta/common/src/main/scala/org/scalameta/internal/logger.scala

```diff
@@ -0,0 +1,95 @@
+// The macro implementations are copied from https://github.com/lihaoyi/sourcecode
+package org.scalameta.internal
+
+import scala.language.existentials
+import scala.language.experimental.macros
+import scala.language.implicitConversions
+
+import scala.reflect.macros.blackbox
+
+abstract class SourceValue[T] {
+  def value: T
+}
+abstract class SourceCompanion[T, V <: SourceValue[T]](build: T => V) {
+  def apply()(implicit s: V): T = s.value
+  implicit def wrap(s: T): V    = build(s)
+}
+
+case class File(value: String) extends SourceValue[String]
+object File extends SourceCompanion[String, File](new File(_)) {
+  implicit def generate: File = macro impl
+
+  def impl(c: blackbox.Context): c.Expr[File] = {
+    import c.universe._
+    val file = c.enclosingPosition.source.path
+    c.Expr[File](q"""${c.prefix}($file)""")
+  }
+}
+
+case class Line(value: Int) extends SourceValue[Int]
+object Line extends SourceCompanion[Int, Line](new Line(_)) {
+  implicit def generate: Line = macro impl
+  def impl(c: blackbox.Context): c.Expr[Line] = {
+    import c.universe._
+    val line = c.enclosingPosition.line
+    c.Expr[Line](q"""${c.prefix}($line)""")
+  }
+}
+
+case class FileLine(file: File, line: Line)
+object FileLine {
+  implicit def generate(implicit file: File, line: Line): FileLine = FileLine(file, line)
+}
+
+case class Text[T](value: T, source: String)
+object Text {
+  implicit def generate[T](v: T): Text[T] = macro Impls.text[T]
```

Comment:
Any idea @xeno-by why this line fails doc generation? 

http://drone.geirsson.com:8001/scalameta/scalameta/170